gdal/frmts/postgisraster/postgisrasterdriver.cpp

```diff
@@ -68,21 +68,11 @@ PostGISRasterDriver::~PostGISRasterDriver() {
  * All connection will be destroyed when the PostGISRasterDriver is destroyed.
  *
  ***************************************************************************/
-PGconn* PostGISRasterDriver::GetConnection(const char* pszConnectionString,
-        const char * pszDbnameIn, const char * pszHostIn, const char * pszPortIn, const char * pszUserIn)
+PGconn* PostGISRasterDriver::GetConnection(const char* pszConnectionString)
 {
     PGconn * poConn = nullptr;
 
-    if( pszHostIn == nullptr ) pszHostIn = "(null)";
-    if( pszPortIn == nullptr ) pszPortIn = "(null)";
-    if( pszUserIn == nullptr ) pszUserIn = "(null)";
-    CPLString osKey = pszDbnameIn;
-    osKey += "-";
-    osKey += pszHostIn;
-    osKey += "-";
-    osKey += pszPortIn;
-    osKey += "-";
-    osKey += pszUserIn;
+    CPLString osKey = pszConnectionString;
```

Comment:
this change is a bit too extreme. The connection string also contains element related to tables that do not determine the PG connection itself. You'd rather want to add an extra parameter for the service name

Reply:
Agree :-)

The key for the internal map is now like this: `service-(null)-(null)-(null)-123456789`